crates/sui-rpc-api/proto/sui/rpc/v2alpha/archival_service.proto

```diff
@@ -0,0 +1,111 @@
+// Copyright (c) Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+syntax = "proto3";
+
+package sui.rpc.v2alpha;
+
+import "google/protobuf/field_mask.proto";
+import "sui/rpc/v2alpha/checkpoint.proto";
+import "sui/rpc/v2alpha/executed_transaction.proto";
+import "sui/rpc/v2alpha/object.proto";
+
+// Alternitive names:
+// - HistoryService
+// - LedgerService
+// - HistoricalService
+// - HistoricalDataService
+service ArchivalService {
+  rpc GetObject(GetObjectRequest) returns (GetObjectResponse);
+  rpc BatchGetObjects(BatchGetObjectsRequest) returns (BatchGetObjectsResponse);
+
+  rpc GetTransaction(GetTransactionRequest) returns (GetTransactionResponse);
+  rpc BatchGetTransactions(BatchGetTransactionsRequest) returns (BatchGetTransactionsResponse);
+
+  rpc GetCheckpoint(GetCheckpointRequest) returns (GetCheckpointResponse);
+  rpc BatchGetCheckpoints(BatchGetCheckpointsRequest) returns (BatchGetCheckpointsResponse);
+}
+
+message GetObjectRequest {
+  // Required. The `ObjectId` of the requested object.
+  optional string object_id = 1;
+
+  // Request a specific version of the object.
+  // If no version is specified, and the object is live, then the latest
+  // version of the object is returned.
+  optional uint64 version = 2;
+
+  // Mask specifying which fields to read.
+  // If no mask is specified, defaults to `object_id,version,digest`.
+  optional google.protobuf.FieldMask read_mask = 4;
+}
+
+message GetObjectResponse {
+  optional Object object = 1;
+}
+
+message BatchGetObjectsRequest {
+  repeated GetObjectRequest requests = 1;
+
+  // Mask specifying which fields to read.
+  // If no mask is specified, defaults to `object_id,version,digest`.
+  optional google.protobuf.FieldMask read_mask = 4;
+}
+
+message BatchGetObjectsResponse {
+  repeated Object objects = 1;
+}
+
+message GetTransactionRequest {
+  // Required. The digest of the requested transaction.
+  optional string digest = 1;
+
+  // Mask specifying which fields to read.
+  // If no mask is specified, defaults to `digest`.
+  optional google.protobuf.FieldMask read_mask = 4;
+}
+
+message GetTransactionResponse {
+  optional ExecutedTransaction transaction = 1;
+}
+
+message BatchGetTransactionsRequest {
+  // Required. The digests of the requested transactions.
+  repeated string digests = 1;
+
+  // Mask specifying which fields to read.
+  // If no mask is specified, defaults to `object_id,version,digest`.
+  optional google.protobuf.FieldMask read_mask = 4;
+}
+
+message BatchGetTransactionsResponse {
+  repeated ExecutedTransaction transactions = 1;
+}
+
+message GetCheckpointRequest {
+  // The sequence number of the requested checkpoint.
+  optional uint64 sequence_number = 1;
+
+  // The digest of the requested checkpoint.
+  repeated string digests = 2;
```

Comment:
Should this look like so?

```suggestion
  optional string digest = 2;
```

Reply:
Yep silly mistake that wasn't caught because the server impl isn't wired up yet